src/engine/threaded_engine.cc

```diff
@@ -391,6 +400,11 @@ void ThreadedEngine::WaitForAll() {
   finished_cv_.wait(lock, [this]() {
       return pending_.load() == 0 || kill_.load();
     });
+  if (global_ex_ptr) {
+    std::exception_ptr ex_ptr = global_ex_ptr;
```

Comment:
use std::move

Reply:
Fixed.

Reply:
Rolling this back, Since the state of global_ex_ptr is not guaranteed to be nullptr after the move and it depends on the implementation. This probably explains why it started failing on windows, after the change. Please let me know if you have any concerns.